sources/core/Stride.Core.Design/IO/UPath.cs

```diff
@@ -290,11 +290,27 @@ public override string ToString()
         /// </summary>
         /// <returns>A string representation of this path in windows form.</returns>
         [NotNull]
+        [Obsolete("Use ToOSPath() instead")]
         public string ToWindowsPath()
         {
             return FullPath.Replace('/', '\\');
         }
 
+        /// <summary>
+        /// Converts this path to a OS path, 
+        /// by replacing each separator with the current operating system
+        /// <see cref="Path.DirectorySeparatorChar"/> 
+        /// </summary>
+        ///<returns>A normalized string path</returns>
+
+        [NotNull]
+        public string ToOSPath()
+        {
+            return FullPath.Replace(
+                    Path.DirectorySeparatorChar == '/' ? '\\' : '/', 
```

Comment:
If the character path is already `/`, isn't it going to still allocate a new string, or is it optimized out?

Reply:
Oh, Indeed, this fact much simplifies the code. Done!